File: DotNet/UserAndGroupCrudSample/UserMethods.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Globalization;
using System.Threading.Tasks;

using RESTAPISDK.Standard;
using RESTAPISDK.Standard.Models;
using RESTAPISDK.Standard.Controllers;
using RESTAPISDK.Standard.Utilities;
using RESTAPISDK.Standard.Exceptions;

namespace UserGroupCrud
{
    class UserMethods : ControllerBase
    {

        public async Task<object> SearchActiveUsers(RESTAPISDKClient p_client)
        {
            UserController userController = p_client.UserController;

            var body = new TspublicRestV2UserSearchRequest();
            body.State = SearchUsersStateEnum.ACTIVE;
            body.OutputFields = new List<string>();
            body.OutputFields.Add("name");

            object result = await userController.SearchUsersAsync(body);
            
            return result;
        }

        public async Task<List<UserResponse>> CreateNewUser(RESTAPISDKClient p_client, List<TspublicRestV2UserCreateRequest> p_users)
        {

            UserController userController = p_client.UserController;

            List<UserResponse> result = new List<UserResponse>();

            foreach (TspublicRestV2UserCreateRequest body in p_users)
            {
                result.Add(await userController.CreateUserAsync(body));
            }

            return result;
        }
    }
}

File: Python/UserAndGroupCrudSample/UserAndGroupCurd/UserMethods.py
#! /usr/bin/env python3
# pylint: disable= no-name-in-module
import json

from restapisdk.models.search_users_state_enum import SearchUsersStateEnum
from restapisdk.models.tspublic_rest_v_2_user_search_request \
    import TspublicRestV2UserSearchRequest

from Python.UserAndGroupCrudSample.UserAndGroupCurd.ControllerBase\
    import ControllerBase


class UserMethods(ControllerBase):
    def searchActiveUsers(self, client):
        v_Name = []
        userController = client.user
        body = TspublicRestV2UserSearchRequest()
        body.state = SearchUsersStateEnum.ACTIVE

        result = userController.search_users(body)
        data_dict = json.loads(result)
        for data in data_dict:
            v_Name.append(data['name'])
        return v_Name

    def CreateNewUser(self, client, userCreateBody):
        userController = client.user
        result = []
        for body in userCreateBody:
            result.append(userController.create_user(body))

        return result
